src/lib/core/config.js

```diff
@@ -299,16 +300,41 @@ Config.prototype.loadContractsConfigFile = function() {
       }
     });
   }
+
   Object.keys(newContractsConfig.contracts).forEach(contractName => {
-      let gas = newContractsConfig.contracts[contractName].gas;
-      let gasPrice = newContractsConfig.contracts[contractName].gasPrice;
-      if (gas && gas.match(unitRegex)) {
-        newContractsConfig.contracts[contractName].gas = utils.getWeiBalanceFromString(gas, web3);
-      }
-      if (gasPrice && gasPrice.match(unitRegex)) {
-        newContractsConfig.contracts[contractName].gasPrice = utils.getWeiBalanceFromString(gasPrice, web3);
-      }
+
+    const gas = newContractsConfig.contracts[contractName].gas;
+    const gasPrice = newContractsConfig.contracts[contractName].gasPrice;
+    const address = newContractsConfig.contracts[contractName].address;
+    const args = newContractsConfig.contracts[contractName].args;
+    const onDeploy = newContractsConfig.contracts[contractName].onDeploy
+    const afterDeploy = newContractsConfig.afterDeploy;
+
+    if (gas && gas.match(unitRegex)) {
+      newContractsConfig.contracts[contractName].gas = utils.getWeiBalanceFromString(gas, web3);
+    }
+
+    if (gasPrice && gasPrice.match(unitRegex)) {
+      newContractsConfig.contracts[contractName].gasPrice = utils.getWeiBalanceFromString(gasPrice, web3);
+    }
+
+    if (address) {
+      newContractsConfig.contracts[contractName].address = extendZeroAddressShorthand(address);
+    }
+
+    if (args && args.length) {
+      newContractsConfig.contracts[contractName].args = args.map(val => typeof val === 'string' ? extendZeroAddressShorthand(val) : val);
```

Comment:
@alaibe I decided to do the string check here and make `extendZeroAddressShorthand` expect always a string